mkt/extensions/tests/test_tasks.py

```diff
@@ -0,0 +1,76 @@
+import mock
+from nose.tools import eq_
+
+from mkt.constants.base import STATUS_PUBLIC
+from mkt.extensions.models import Extension, ExtensionVersion
+from mkt.extensions.tasks import fetch_icon
+from mkt.site.tests import TestCase
+
+
+class TestFetchIcon(TestCase):
+    def setUp(self):
+        super(TestFetchIcon, self).setUp()
+        self.extension = Extension.objects.create()
+        self.version = self.extension.versions.create(manifest={
+            'icons': {
+                '128': '/path/to/icon.png'
+            }
+        }, version='0.1')
+
+    @mock.patch('mkt.extensions.tasks.save_icon')
+    @mock.patch('mkt.extensions.tasks.ZipFile')
+    @mock.patch('mkt.extensions.tasks.private_storage')
+    def _test(self, private_storage_mock, ZipFile_mock, save_icon_mock,
```

Comment:
I agree - I spent a lot of time running around trying to find a better way of doing this, and this is the best I could come up with. I tried to keep the calls and therefore the mocks simple and heavily document + test the various cases to balance the awkwardness of the mocks.